reconcile/utils/aws_api.py

```diff
@@ -1155,6 +1167,38 @@ def get_tgws_details(self, account, region_name, routes_cidr_block,
 
         return results
 
+    @staticmethod
+    def _get_hosted_zone_id(zone: HostedZoneTypeDef) -> str:
+        # 'Id': '/hostedzone/THISISTHEZONEID'
+        return zone["Id"].split("/")[-1]
+
+    def _get_hosted_zone_record_sets(self,
+                                     route53: Route53Client,
+                                     zone_name: str) -> List[ResourceRecordSetTypeDef]:
+        zones = route53.list_hosted_zones_by_name(DNSName=zone_name)["HostedZones"]
+        if not zones:
+            return []
+        zone_id = self._get_hosted_zone_id(zones[0])
+        return route53.list_resource_record_sets(HostedZoneId=zone_id)["ResourceRecordSets"]
+
+    @staticmethod
+    def _filter_record_sets(record_sets: List[ResourceRecordSetTypeDef],
+                            zone_name: str,
+                            zone_type: str) -> List[ResourceRecordSetTypeDef]:
+        return [r for r in record_sets if r["Name"] == f"{zone_name}." and r["Type"] == zone_type]
+
+    @staticmethod
+    def _extract_records(resource_records: List[ResourceRecordTypeDef]) -> list[str]:
+        # [{'Value': 'ns.example.com.'}, ...]
+        return [r['Value'].rstrip(".") for r in resource_records]
+
+    def get_route53_zone_ns_records(self, account_name, zone_name, region):
+        route53 = self._account_route53_client(account_name, region)
+        record_sets = self._get_hosted_zone_record_sets(route53, zone_name)
+        resource_records = self._filter_record_sets(record_sets, zone_name, "NS")[0]["ResourceRecords"]
```

Comment:
leaving this comment for other reviewers: i was first puzzled about picking just the first NS record but then i learned what a record set is. _filter_record_sets returns a "set of records for a zone and a record type", in this case it returns all NS records for the zone at hand. so this is absolutely fine and i started my learning path for route 53